### PR TITLE
chore(nightly): fix targetEnv schema

### PR DESCRIPTION
**Pull request by fu050409**
TypeScript static type check unable to auto check dyn types.

Declare as `z.enum(Object.values(TARGET_ENV) as [string, ...string[]]).optional(),` will be seen as `z.ZodOptional<z.ZodEnum<[string, ...string[]]>>`.

Use `(typeof TARGET_ENV)[keyof typeof TARGET_ENV]` will fix this issue.

### CODE REVIEW

File: packages/core/src/config/schema.ts [hunk 1]
```diff
@@ -50,9 +51,7 @@ const outputSchema = z
     path: z.string().optional(),
     publicPath: z.string().optional(),
     assetsFilename: z.string().optional(),
-    targetEnv: z
-      .enum(Object.values(TARGET_ENV) as [string, ...string[]])
-      .optional(),
+    targetEnv: z.custom<(typeof TARGET_ENV)[keyof typeof TARGET_ENV]>(),
```
Comment (ErKeLost) [inline packages/core/src/config/schema.ts:54]: z.custom 是不是需要返回一个函数

Comment (ErKeLost) [reply packages/core/src/config/schema.ts:54]: https://zod.dev/?id=custom-schemas

Comment (fu050409) [reply packages/core/src/config/schema.ts:54]: 用法就是`z.custom<T>()`，不过这里忘了加`optional`了我改一下

Comment (fu050409) [reply packages/core/src/config/schema.ts]: > https://zod.dev/?id=custom-schemas

哦哦是需要一个验证函数是吧

Comment (ErKeLost) [reply packages/core/src/config/schema.ts]: 对 我看他说要最好加上这个函数

**Pull request merged**